Assets/Scripts/Modules/Combat/ICombatModule.cs

```diff
@@ -1,8 +1,19 @@
 ﻿/*
  * Authors: Ian Macpherson, Isaiah Mann
- * Description: An interface to handle the turn logic
+ * Description: An interface to handle the combat logic
  */
 
 public interface ICombatModule {
 
+	bool IsTargetInRange (IUnit unit);
+
+	bool IsTargetAdjacent (IUnit unit);
+
+	void MeleeAttack (IUnit unit);
+
+	void RangedAttack (IUnit unit);
+
+	void FleeAttempt (IStatModule playerstats, IUnit unit);
+
+	void KillMonster (IUnit unit);
```

Comment:
Can this be "KillUnit" because it can conceivably also be applied to the player. You can also include KillPlayer and KillMonster, but I think we should a more general version.